hooks/tk-multi-publish2/basic/publish_lookfiles.py

```diff
@@ -0,0 +1,426 @@
+        item.properties["publish_path"] = publish_path
+        item.properties["path"] = path
+        if os.path.exists(publish_path):
+            raise Exception("The file '{}' has already been copied to the publish location.".format(path))
+
+        return True
+
+    def publish(self, settings, item):
+        """
+        Executes the publish logic for the given item and settings.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+        publish_path = item.properties["publish_path"]
+        publish_dir = os.path.dirname(publish_path)
+
+        self.logger.debug("Making directory: '{}'".format(publish_dir))
+        engine = sgtk.platform.current_engine()
+        engine.ensure_folder_exists(publish_dir)
+
+        item.properties["publish_type"] = "Katana Look File"
+
+        super(KatanaLookFilePublishPlugin, self).publish(settings, item)
+
+    def create_settings_widget(self, parent):
+        """
+        Creates a Qt widget, for the supplied parent widget (a container widget
+        on the right side of the publish UI).
+
+        :param parent: The parent to use for the widget being created
+        :return: A QtGui.QWidget or subclass that displays information about
+            the plugin and/or editable widgets for modifying the plugin's
+            settings.
+        """
+        description_widget = super(KatanaLookFilePublishPlugin, self).create_settings_widget(None)
+        return LookFilePublishWidget(parent=parent, description_widget=description_widget) 
+
+    def get_ui_settings(self, widget):
+        """
+        Invoked by the Publisher when the selection changes. This method gathers the settings
+        on the previously selected task, so that they can be later used to repopulate the
+        custom UI if the task gets selected again. They will also be passed to the accept, validate,
+        publish and finalize methods, so that the settings can be used to drive the publish process.
+
+        The widget argument is the widget that was previously created by
+        `create_settings_widget`.
+
+        The method returns a dictionary, where the key is the name of a
+        setting that should be updated and the value is the new value of that
+        setting. Note that it is up to you how you want to store the UI's state as
+        settings and you don't have to necessarily to return all the values from
+        the UI. This is to allow the publisher to update a subset of settings
+        when multiple tasks have been selected.
+
+        Example::
+
+            {
+                 "setting_a": "/path/to/a/file"
+            }
+
+        :param widget: The widget that was created by `create_settings_widget`
+        """
+        return widget.getData()
+
+    def set_ui_settings(self, widget, settings):
```

Comment:
decorate it as `@staticmethod`?
```suggestion
    @staticmethod
    def set_ui_settings(widget, settings):
```

Reply:
this is also an overridden method